app/controllers/concerns/ad_renderable.rb

```diff
@@ -0,0 +1,41 @@
+module AdRenderable
+  extend ActiveSupport::Concern
+
+  included do
+    attr_reader :template_name, :theme_name
+  end
```

Comment:
These `attr_readers` may work declared in the module without needing to declare inside the `included` method.